starknet-foundry/crates/cheatable-starknet/src/rpc.rs

```diff
@@ -0,0 +1,516 @@
+                    )
+                    .unwrap();
+                let orginal_block_info = vm
+                    .get_continuous_range((block_info_ptr + 1_usize).unwrap() as Relocatable, 2)
+                    .unwrap();
+                vm.load_data(ptr_cheated_block_info_c, &orginal_block_info)
+                    .unwrap();
+
+                // create a new segment with replaced execution_info including pointer to updated block info
+                let ptr_cheated_exec_info_c = vm
+                    .load_data(
+                        ptr_cheated_exec_info,
+                        &vec![MaybeRelocatable::RelocatableValue(ptr_cheated_block_info)],
+                    )
+                    .unwrap();
+                let original_execution_info = vm
+                    .get_continuous_range((execution_info_ptr + 1_usize).unwrap() as Relocatable, 4)
+                    .unwrap();
+                vm.load_data(ptr_cheated_exec_info_c, &original_execution_info)
+                    .unwrap();
+
+                let remaining_gas = gas_counter - constants::GET_EXECUTION_INFO_GAS_COST;
+                let response = GetExecutionInfoResponse {
+                    execution_info_ptr: ptr_cheated_exec_info,
+                };
+                let response_w = SyscallResponseWrapper::Success {
+                    gas_counter: remaining_gas,
+                    response,
+                };
+                response_w.write(vm, &mut self.syscall_handler.syscall_ptr)?;
+                return Ok(());
+            }
+        }
+        self.syscall_handler.execute_next_syscall(vm, hint)
+    }
+
+    fn verify_syscall_ptr(&self, actual_ptr: Relocatable) -> SyscallResult<()> {
+        if actual_ptr != self.syscall_handler.syscall_ptr {
+            return Err(SyscallExecutionError::BadSyscallPointer {
+                expected_ptr: self.syscall_handler.syscall_ptr,
+                actual_ptr,
+            });
+        }
+
+        Ok(())
+    }
+}
+
+/// Executes a specific call to a contract entry point and returns its output.
+fn execute_entry_point_call_cairo1(
+    call: CallEntryPoint,
+    contract_class: &ContractClassV1,
+    state: &mut dyn State,
+    cheated_state: &CheatedState,
+    resources: &mut ExecutionResources,
+    context: &mut EntryPointExecutionContext,
+) -> EntryPointExecutionResult<CallInfo> {
+    let VmExecutionContext {
+        mut runner,
+        mut vm,
+        mut syscall_handler,
+        initial_syscall_ptr,
+        entry_point,
+        program_segment_size,
+    } = initialize_execution_context(call, contract_class, state, resources, context)?;
+
+    let args = prepare_call_arguments(
+        &syscall_handler.call,
+        &mut vm,
+        initial_syscall_ptr,
+        &mut syscall_handler.read_only_segments,
+        &entry_point,
+    )?;
+    let n_total_args = args.len();
+
+    // Fix the VM resources, in order to calculate the usage of this run at the end.
```

Comment:
By `fix` you mean `store` right?

Reply:
Fix, means make it constant at the moment so you can subtract later. Maybe snapshot would be a better wording

Reply:
[Review 2](https://github.com/foundry-rs/starknet-foundry/pull/181/commits/177b7e3da6761ffb59c65ba1bb3b0f81211af4e2)